src/main/scala/com/cognite/sdk/scala/v1/fdm/instances/InstancePropertyValue.scala

```diff
@@ -110,32 +112,13 @@ object InstancePropertyValue {
         }
         numericInstantPropType.map(Right(_))
       case v if v.isBoolean => v.asBoolean.map(s => Right(InstancePropertyValue.Boolean(s)))
-      case v if v.isObject =>
-        val res: Option[Either[DecodingFailure, InstancePropertyValue]] = v.asObject.map(obj =>
-          if (obj.contains("externalId") && obj.contains("space") && obj.size === 2) {
-            for {
-              spaceJson <- obj("space").toRight(DecodingFailure("missing space", c.history))
-              space <- spaceJson.asString.toRight(DecodingFailure("space isn't string", c.history))
-              externalIdJson <- obj("externalId").toRight(
-                DecodingFailure("missing externalId", c.history)
-              )
-              externalId <- externalIdJson.asString.toRight(
-                DecodingFailure("externalId isn't string", c.history)
-              )
-            } yield InstancePropertyValue.ViewDirectNodeRelation(
-              Option(
-                DirectRelationReference(
-                  space,
-                  externalId
-                )
-              )
-            )
-          } else {
-            Right(InstancePropertyValue.Object(v))
-          }
+      case v if v.isObject && isDirectRelation(v) =>
```

Comment:
```suggestion
      case v if isDirectRelation(v) =>
```
should work like that too, not sure which one reads better though